Bottom: 66b645c3c06e5f84fec7c4b5eb962bfb50fd75f4
Top:    0c94e161419370c7dd6fc79d28109deca86fa969
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:17:44 -0500

Add environment variables for parameters


---
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index 412ac9a5..7e6c0cc2 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -33,6 +33,9 @@ config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
 # environment passed to the container
 container_env:
   BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
+  BUILDBOT_SMTP_HOST: 'relay.dnsexit.com'
+  BUILDBOT_SMTP_USER: 'Dean.cox'
+  BUILDBOT_SMTP_PASS: 'no5156'
   BUILDBOT_WEB_PORT: '{{buildbot_http_port}}'
   BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
   BUILDBOT_WORKER_PORT: '{{buildbot_worker_port}}'
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index e17bce26..961819c3 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -20,7 +20,7 @@ c['change_source'] = []
 # Defines the TCP port to listen on for connections from workers.
 # This must match the value configured into the buildworkers (with their
 # --master option)
-c['protocols'] = {"pb": {"port": os.getenv('BUILDBOT_WORKER_PORT', 9989)}}
+c['protocols'] = {"pb": {"port": int(os.getenv('BUILDBOT_WORKER_PORT', 9989))}}
 
 # STATUS TARGETS
 
@@ -40,13 +40,17 @@ mn = reporters.MailNotifier(
     fromaddr="buildbot@novatechweb.com",
     sendToInterestedUsers=True,
     extraRecipients=["andrew.cooper@novatechweb.com"],
-    useTls=True, relayhost="relay.dnsexit.com", smtpPort=26,
-    smtpUser="Dean.cox", smtpPassword="no5156")
+    useTls=True,
+    relayhost=os.getenv('BUILDBOT_SMTP_HOST'),
+    smtpPort=int(os.getenv('BUILDBOT_SMTP_PORT')),
+    smtpUser=os.getenv('BUILDBOT_SMTP_USER'),
+    smtpPassword=os.getenv('BUILDBOT_SMTP_PASS')
+)
 c['services'].append(mn)
 
 # WEB SERVER
 c['www'] = {
-    'port': os.getenv('BUILDBOT_WEB_PORT', 8010),
+    'port': int(os.getenv('BUILDBOT_WEB_PORT', 8010)),
     'plugins': {
         'waterfall_view': True,
         'console_view': True,
@@ -85,7 +89,7 @@ c['buildbotURL'] = os.getenv('BUILDBOT_WEB_URL', "http://localhost:%s/" % (c['ww
 c['db'] = {
     # This specifies what database buildbot uses to store its state.  You can leave
     # this at its default for all but the largest installations.
-    'db_url': os.getenv('BUILDBOT_DATABASE', "sqlite://state.sqlite"),
+    'db_url' : os.getenv('BUILDBOT_DATABASE', "sqlite://"),
 }
 
 # Extra Options
